packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -118,6 +118,7 @@ class FlutterOptions {
   static const String kDeferredComponents = 'deferred-components';
   static const String kAndroidProjectArgs = 'android-project-arg';
   static const String kInitializeFromDill = 'initialize-from-dill';
+  static const String kFatalLogWarnings = 'fatal-log-warnings';
```

Comment:
Googling for precedents for a flag like this, I found that "fatal-log-warnings" gets literally zero results. We might want to consider using one of the arguments that other tools use. It seems gcc uses "-Werror", which doesn't really fit our style. `as` (the assembler) uses `-fatal-warnings`, llvm seems to use `--fatal-warnings`. Scalar uses `-Xfatal-warnings`. Mozilla once internally used `FAIL_ON_WARNINGS`. Microsoft's C# compiler uses `-warnaserror`. Solaris' C compiler uses `-errwarn`. Dart's VM compiler has `--warning_as_error`. Dart's JS compiler has `--fatal-warnings`. 

I couldn't find any uses of my personal preference `--warnings-are-fatal`, FWIW. Looks like the most common convention is `--fatal-warnings`.

Reply:
OK, renamed to `--fatal-warnings`, that works for me.